rasa/train.py

```diff
@@ -157,6 +209,20 @@ async def _train_async_internal(
         file_importer.get_stories(), file_importer.get_nlu_data()
     )
 
+    # get the fingerprint
+    new_fingerprint = await model.model_fingerprint(file_importer)
+    old_model = model.get_latest_model(output_path)
+
+    fingerprint_comparison = model.should_retrain(
+        new_fingerprint, old_model, train_path, force_training
+    )
+
+    if dry_run:
+        code, texts = dry_run_result(fingerprint_comparison)
+        for text in texts:
+            print_warning(text) if code > 0 else print_success(text)
+        sys.exit(code)
```

Comment:
This might be a real PITA, but I wonder if the code should be passed up to `rasa/cli/train.py` and exiting handled there? Just as `rasa/cli/` where we tend to handle things like this.